### PR TITLE
Rename System.Threading.Thread.m_ManagedThreadId to System.Threading.Thread._ManagedThreadId to align with CoreRT implementation

### PR DESCRIPTION
**Pull request by cshung**

### CODE REVIEW

**Comment by cshung** [comment]
@kouvel

File: src/mscorlib/src/System/Threading/Thread.cs [hunk 1]
```diff
@@ -128,7 +128,12 @@ internal sealed class Thread : RuntimeThread
 
         private IntPtr DONT_USE_InternalThread;        // Pointer
         private int m_Priority;                     // INT32
-        private int m_ManagedThreadId;              // INT32
+
+        // The following field is required for interop with the VS Debugger
+        // Prior to making any changes to this field, please reach out to the VS Debugger 
+        // team to make sure that your changes are not going to prevent the debugger
+        // from working.
+        private int _ManagedThreadId;              // INT32
```
Comment (jkotas) [inline src/mscorlib/src/System/Threading/Thread.cs]: Shouldn't this be lower case `_managedThreadId` ?

Comment (jkotas) [inline src/mscorlib/src/System/Threading/Thread.cs]: Also, it is actually `class` in CoreRT... .

File: src/mscorlib/src/System/Threading/Thread.cs [hunk 2]
```diff
@@ -200,7 +205,7 @@ internal Thread(ParameterizedThreadStart start, int maxStackSize)
 
         public override int GetHashCode()
         {
-            return m_ManagedThreadId;
+            return _ManagedThreadId;
```
Comment (jkotas) [inline src/mscorlib/src/System/Threading/Thread.cs]: You have missed this place... .

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Pull request merged**